packages/core/src/rules/common/missing-required-schema-properties.ts

```diff
@@ -0,0 +1,30 @@
+import { Oas3Rule } from '../../visitors';
+import { Oas3Schema, Oas3_1Schema } from '../../typings/openapi';
+import { Oas2Schema } from 'core/src/typings/swagger';
+import { UserContext } from 'core/src/walk';
+
+export const MissingRequiredSchemaProperties: Oas3Rule = () => {
+  return {
+    Schema: {
+      enter(schema: Oas3Schema | Oas3_1Schema | Oas2Schema, { location, report }: UserContext) {
+        if (schema?.required) {
+          const missingRequiredProperties: Array<string> = schema.required.filter((property) => {
+            return !schema.properties || !Object.keys(schema.properties!).includes(property);
+          });
+
+          if (missingRequiredProperties.length) {
+            const reportMessage =
+              missingRequiredProperties.length > 1
+                ? `Properties: ${missingRequiredProperties.join(', ')} are required.`
+                : `Property ${missingRequiredProperties.join()} is required.`;
```

Comment:
I wonder if these messages are clear enough.

```
required:
  - id
  - pet
  - test
properties:
  id:
     type: string
  pet:
     type: string
```
Test is not explicitly defined (although allowed through `additionalProperties`). I think we're stating that the problem is: `Required property is undefined.`

```suggestion
                ? `Required properties are undefined: ${missingRequiredProperties.join(', ')}.`
                : `Required property ${missingRequiredProperties.join()} is undefined.`;
```
What do you think about this suggestion?